plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/Session.kt

```diff
@@ -114,15 +114,40 @@ class Session(val tabID: String, val project: Project) {
     /**
      * Triggered by the Insert code follow-up button to apply code changes.
      */
-    suspend fun insertChanges(
+    suspend fun insertChangesAndUpdateFileComponents(
```

Comment:
I'm going to add comments to make sure I understand the roles of each component. 

This is used to 
1. write files to disk and log any references
2. update the file tree to present visually what has been added or not

Is that right?